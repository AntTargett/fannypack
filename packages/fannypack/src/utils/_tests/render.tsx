import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
