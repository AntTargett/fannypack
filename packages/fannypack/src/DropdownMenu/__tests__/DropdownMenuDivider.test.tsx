import * as React from 'react';
import { DropdownMenu } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">{menu => <DropdownMenu.Divider {...menu} />}</DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for CSS props', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Divider {...menu} backgroundColor="red" />}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('DropdownMenu.Divider.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu.Divider
            {...menu}
            overrides={{ DropdownMenu: { Divider: { css: { root: { backgroundColor: 'red' } } } } }}
          />
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('DropdownMenu.Divider.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">{menu => <DropdownMenu.Divider {...menu} />}</DropdownMenu.State>,
      { theme: { DropdownMenu: { Divider: { css: { root: { backgroundColor: 'red' } } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">{menu => <DropdownMenu.Divider {...menu} />}</DropdownMenu.State>,
      { theme: { DropdownMenu: { Divider: { defaultProps: { className: 'test', color: 'primary' } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
