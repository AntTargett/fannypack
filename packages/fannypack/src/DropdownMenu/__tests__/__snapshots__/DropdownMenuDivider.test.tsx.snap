// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  border: 1px solid #e6e6e6;
  border-width: 0 0 1px 0;
  margin: 1em 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-1[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  margin: 0 1em;
  width: 0;
  height: auto;
}

.emotion-0.emotion-0 {
  margin: 0.5rem 0;
}

.emotion-3 {
  color: #3722ed !important;
}

<div
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 emotion-3 test fp-Box fp-Divider fp-DropdownMenuDivider"
  role="separator"
/>
`;

exports[`overrides DropdownMenu.Divider.css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  border: 1px solid #e6e6e6;
  border-width: 0 0 1px 0;
  margin: 1em 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-1[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  margin: 0 1em;
  width: 0;
  height: auto;
}

.emotion-0.emotion-0 {
  margin: 0.5rem 0;
}

.emotion-0 {
  background-color: red;
}

<div
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Divider fp-DropdownMenuDivider"
  role="separator"
/>
`;

exports[`props should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  border: 1px solid #e6e6e6;
  border-width: 0 0 1px 0;
  margin: 1em 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-1[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  margin: 0 1em;
  width: 0;
  height: auto;
}

.emotion-0.emotion-0 {
  margin: 0.5rem 0;
}

<div
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Divider fp-DropdownMenuDivider"
  role="separator"
/>
`;

exports[`props should render correctly for CSS props 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  border: 1px solid #e6e6e6;
  border-width: 0 0 1px 0;
  margin: 1em 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-1[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  margin: 0 1em;
  width: 0;
  height: auto;
}

.emotion-0.emotion-0 {
  margin: 0.5rem 0;
}

.emotion-3 {
  background-color: red !important;
}

<div
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Divider fp-DropdownMenuDivider"
  role="separator"
/>
`;

exports[`theming DropdownMenu.Divider.css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  border: 1px solid #e6e6e6;
  border-width: 0 0 1px 0;
  margin: 1em 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-1[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  margin: 0 1em;
  width: 0;
  height: auto;
}

.emotion-0.emotion-0 {
  margin: 0.5rem 0;
}

.emotion-0 {
  background-color: red;
}

<div
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Divider fp-DropdownMenuDivider"
  role="separator"
/>
`;
